settings_test.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/kubewarden/container-resources-policy/resource"
+	kubewarden_protocol "github.com/kubewarden/policy-sdk-go/protocol"
+)
+
+func TestParsingSettingsWithNoValueProvided(t *testing.T) {
+	rawSettings := []byte(`{}`)
+	settings := &Settings{}
+	if err := json.Unmarshal(rawSettings, settings); err != nil {
+		t.Fatalf("Unexpected error %+v", err)
+	}
+	valid, _ := settings.Valid()
+	if valid {
+		t.Fatal("settings reported as valid. User must set some required fields")
+	}
+}
+
+func TestParsingSettings(t *testing.T) {
+	var tests = []struct {
+		name         string
+		rawSettings  []byte
+		errorMessage string
+	}{
+		{"valid settings", []byte(`{"maxCpuLimit": "1m", "maxMemoryLimit": "1G", "maxCpuRequest": "1m", "maxMemoryRequest": "1G", "ignoreImages": ["image:latest"]}`), ""},
+		{"no suffix", []byte(`{"maxCpuLimit": "1", "maxMemoryLimit": "2","maxCpuRequest": "1", "maxMemoryRequest": "2"}`), ""},
+		{"invalid cpu limit suffix", []byte(`{"maxCpuLimit": "1x", "maxMemoryLimit": "2", "maxCpuRequest": "1m", "maxMemoryRequest": "1G"}`), "failed to parse cpu limit quantity"},
+		{"invalid memory limit suffix", []byte(`{"maxCpuLimit": "1", "maxMemoryLimit": "2x", "maxCpuRequest": "1m", "maxMemoryRequest": "1G"}`), "failed to parse memory limit quantity"},
+		{"invalid cpu request suffix", []byte(`{"maxCpuLimit": "1", "maxMemoryLimit": "2", "maxCpuRequest": "1x", "maxMemoryRequest": "2G"}`), "failed to parse cpu request quantity"},
+		{"invalid memory request suffix", []byte(`{"maxCpuLimit": "1", "maxMemoryLimit": "2", "maxCpuRequest": "1m", "maxMemoryRequest": "2x"}`), "failed to parse memory request quantity"},
+		{"invalid settings", []byte(`{}`), "no settings provided. At least one resource limit or request must be verified"},
```

Comment:
With this case, I would be ok dropping `TestParsingSettingsWithNoValueProvided()`.